src/UnityExtension/Assets/Editor/GitHub.Unity/UI/PUblishView.cs

```diff
@@ -127,111 +136,111 @@ public override void OnGUI()
 
-            GUILayout.BeginHorizontal();
-            {
                 GUILayout.FlexibleSpace();
-                GUI.enabled = !string.IsNullOrEmpty(repoName) && !isBusy;
-                if (GUILayout.Button(PublishViewCreateButton))
-                {
-                    isBusy = true;
 
-                    var organization = owners[selectedOwner] == username ? null : owners[selectedOwner];
-
-                    Client.CreateRepository(new NewRepository(repoName)
-                    {
-                        Private = togglePrivate,
-                    }, (repository, ex) =>
+                GUILayout.BeginHorizontal();
+                {
+                    GUILayout.FlexibleSpace();
+                    EditorGUI.BeginDisabledGroup(!IsFormValid);
+                    if (GUILayout.Button(PublishViewCreateButton))
                     {
-                        Logger.Trace("Create Repository Callback");
+                        isBusy = true;
 
-                        if (ex != null)
-                        {
-                            error = ex.Message;
-                            isBusy = false;
-                            return;
-                        }
+                        var organization = owners[selectedOwner] == username ? null : owners[selectedOwner];
 
-                        if (repository == null)
+                        Client.CreateRepository(new NewRepository(repoName)
                         {
-                            Logger.Warning("Returned Repository is null");
-                            isBusy = false;
-                            return;
-                        }
+                            Private = togglePrivate,
+                        }, (repository, ex) =>
+                        {
+                            Logger.Trace("Create Repository Callback");
+
+                            if (ex != null)
+                            {
+                                error = ex.Message;
+                                isBusy = false;
+                                return;
+                            }
+
+                            if (repository == null)
+                            {
+                                Logger.Warning("Returned Repository is null");
+                                isBusy = false;
+                                return;
+                            }
 
-                        GitClient.RemoteAdd("origin", repository.CloneUrl)
-                                 .Then(GitClient.Push("origin", Repository.CurrentBranch.Value.Name))
-                                 .ThenInUI(Parent.Finish)
-                                 .Start();
-                    }, organization);
+                            GitClient.RemoteAdd("origin", repository.CloneUrl)
+                                     .Then(GitClient.Push("origin", Repository.CurrentBranch.Value.Name))
+                                     .ThenInUI(Parent.Finish)
```

Comment:
Before doing this, `isBusy` should be reset to false. We shouldn't be calling `Parent.Finish` here anyway, we should be calling `Finish` (which calls `Parent.Finish`), and then we can reset the busy flag there.